prov/util/src/util_shm.c

```diff
@@ -267,7 +267,7 @@ int smr_create(const struct fi_provider *prov, struct smr_map *map,
 
 	(*smr)->map = map;
 	(*smr)->version = SMR_VERSION;
-	(*smr)->flags = SMR_FLAG_ATOMIC | SMR_FLAG_DEBUG;
+	(*smr)->flags = SMR_FLAG_ATOMIC | SMR_FLAG_DEBUG | attr->flags;
```

Comment:
It is odd to pass in flags via the attributes, but also have the code assume certain flags should be set.

While looking into this, it's confusing that the ATOMIC and DEBUG flags are sometimes only defined.  That's normal for an ifdef type of check, but for a flag it's unusual.  It means that code such as:
```
if ((flags & DEBUG) == DEBUG)
```
returns true, even though no flag is actually set.

Reply:
@shefty Agreed that is weird and shouldn't be used like that. I can rethink this.

Not sure I understand what you mean by the flags are sometimes only defined. From what I can tell, they are always defined, just always correlate to how libfabric is configured/built.
```
#ifdef HAVE_ATOMICS
#define SMR_FLAG_ATOMIC	(1 << 0)
#else
#define SMR_FLAG_ATOMIC	(0 << 0)
#endif

#if ENABLE_DEBUG
#define SMR_FLAG_DEBUG	(1 << 1)
#else
#define SMR_FLAG_DEBUG	(0 << 1)
#endif
```

Reply:
SMR_FLAG_ATOMIC and SMR_FLAG_DEBUG can both be defined as 0.  0 isn't really a flag value.